### ISSUE TITLE
Overwriting global Promise.prototype

### ISSUE THREAD

**Issue by gchamon** [handsontable/handsontable#5398]
In this code section in `dist/handsontable.js`
```
  Internal.prototype = __webpack_require__(62)($Promise.prototype, {
    // 25.4.5.3 Promise.prototype.then(onFulfilled, onRejected)
    then: function then(onFulfilled, onRejected) {
      var reaction = newPromiseCapability(speciesConstructor(this, $Promise));
      reaction.ok = typeof onFulfilled == 'function' ? onFulfilled : true;
      reaction.fail = typeof onRejected == 'function' && onRejected;
      reaction.domain = isNode ? process.domain : undefined;
      this._c.push(reaction);
      if (this._a) this._a.push(reaction);
      if (this._s) notify(this, false);
      return reaction.promise;
    },
    // 25.4.5.1 Promise.prototype.catch(onRejected)
    'catch': function (onRejected) {
      return this.then(undefined, onRejected);
    }
  });
```
`$Promise.prototype` is rewritten. In my application, `$Promise` is actually pointing to `window["Promise"]` and the rewrite is affecting everything globally, removing the method `finally` that is used extensively in my code.

I believe this is a webpack doing, but you guys should be aware of that, in case you are not already.

**Comment by gchamon** [comment]
Am I correct to assume this code is generated by webpack? If not, a simple
```
'finally': Promise.prototype.finally // (or Promise.prototype.then, though it defeats the purpose and semantics of having fianally)
```
will suffice (I have modified the file and tested with success).
Another thing to notice ``babel-polyfill` is outdated. `@babel/polyfill` it at version `7.0.0`. I think it might also be worth taking a look.

**Comment by selo796** [comment]
> Am I correct to assume this code is generated by webpack? If not, a simple
> 
> ```
> 'finally': Promise.prototype.finally // (or Promise.prototype.then, though it defeats the purpose and semantics of having finally)
> ```
> will suffice (I have modified the file and tested with success).
> Another thing to notice `babel-polyfill` is outdated. `@babel/polyfill` it at version `7.0.0`. I think it might also be worth taking a look.

Hi @gchamon , 
the code is generated by webpack as far I can understand. I would like to also try your solution. Which file did you modified?

**Comment by gchamon** [comment]
I used `dist/handsontable.js` directly, adding the code `'finally': Promise.prototype.finally` at the end of the offending code section, which starts with the line `Internal.prototype = __webpack_require__(62)($Promise.prototype, {`. The result is:

```
  Internal.prototype = __webpack_require__(62)($Promise.prototype, {
    // 25.4.5.3 Promise.prototype.then(onFulfilled, onRejected)
    then: function then(onFulfilled, onRejected) {
      var reaction = newPromiseCapability(speciesConstructor(this, $Promise));
      reaction.ok = typeof onFulfilled == 'function' ? onFulfilled : true;
      reaction.fail = typeof onRejected == 'function' && onRejected;
      reaction.domain = isNode ? process.domain : undefined;
      this._c.push(reaction);
      if (this._a) this._a.push(reaction);
      if (this._s) notify(this, false);
      return reaction.promise;
    },
    // 25.4.5.1 Promise.prototype.catch(onRejected)
    'catch': function (onRejected) {
      return this.then(undefined, onRejected);
    },
    'finally': Promise.prototype.finally
  });
```
The downside is that you can't use the minified version. Also, compiling the code rewrites the section.

I couldn't find traces of the offending code in the minified file, but that doesn't mean it isn't there.

**Comment by gchamon** [comment]
I can't say, since I use Linux, but Chrome has some sort of fail-safe that prevents this overwriting. Firefox is affected, though.

**Comment by gchamon** [comment]
Here is polyfill implementation: https://github.com/es-shims/Promise.prototype.finally

reference: http://2ality.com/2017/07/promise-prototype-finally.html

**Comment by jfcherng** [comment]
Thanks for filing this issue so that I finally found the culprit who made some of my pages broken in IE/Edge.

I am using Webpack + npm and do not want to modify Handsontable's source codes so I simply overwrite the global/window Promise back again with core-js' implementation.

```javascript
global.Handsontable = require('handsontable');

// Fix "Promise" is overwritten by handsontable
// @see: https://github.com/handsontable/handsontable/issues/5398
global.Promise = require('core-js/fn/promise');
```

**Comment by wszymanski** [comment]
Maybe helpful: https://babeljs.io/docs/en/v7-migration

**Comment by trickstival** [comment]
If you're using handsontable 2.0.0 with npm it's possible to use the workaround mentioned by @gchamon by writing a sed command inside the postinstall script in your package.json:
```json
~> package.json
{
    "scripts": {
        "postinstall": "sed -i '/25.4.5.1 Promise.prototype.catch(onRejected)/i finally: Promise.prototype.finally', node_modules/handsontable/dist/handsontable.js"
    }
}
```

Tested with vue-handsontable

**Comment by acontreras89** [comment]
Can we get a 5.x version including this fix?

**Comment by AMBudnik** [comment]
Hi @acontreras89 

> this issue has been fixed in v 6.2.1

you'd need to update to 6.2.1 or newer version to get this fix.
